BestFloww/BestFlow
# updated jest config and added dao/interface for intents

## Pull Request Summary
Added dao/interface for intents. Also made schema with useful methods.

### Change types



### Testing

I do not think we can properly test our schema. :(

### How to manually test this PR (QA instructions)
I do not think this is manually testable **yet**. Once the rest of the BE tickets are done we can do a run through at the URL to see if it works. Please CAREFULLY review all the code her and make sure tests are robust. In particular, pay attention to schema code.

You can also checkout this branch and run ```npm test```

## Questions
What do you think of the general design patterns used here?
